eng/pipelines/templates/steps/npm-release-task.yml

```diff
@@ -0,0 +1,56 @@
+parameters:
+  Registry: ''
+
+steps:
+  - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+  - ${{if eq(parameters.Registry, 'https://registry.npmjs.org/')}}:
+      - task: PowerShell@2
+        displayName: Get package list
+        inputs:
+          targetType: filePath
+          filePath: eng/tools/publish-to-npm.ps1
+          arguments: -pathToArtifacts $(Package.Archive) -accessLevel "public" -tag "$(Tag)" -additionalTag "$(AdditionalTag)" -registry ${{parameters.Registry}} -npmToken $(azure-sdk-npm-token)
+          pwsh: true
+        condition: succeeded()
+      - ${{ each package in split(variables.NpmPublishList, ',') }}:
+          - pwsh: write-host ${{package}}; write-host ${{variables.NpmPublishList}}
+          - task: EsrpRelease@6
+            inputs:
+              displayName: 'Publish ${{package}} to ESRP'
+              ConnectedServiceName: 'ESRP Release Service'
+              Intent: 'PackageDistribution'
+              ContentType: 'npm'
+              FolderLocation: ${{package}}
+              Owners: $(Build.RequestedForEmail)
+              Approvers: 'wesh@microsoft.com'
+              ServiceEndpointUrl: 'https://api.esrp.microsoft.com'
+              MainPublisher: 'ESRPRELPACMANTEST'
+              DomainTenantId: '72f988bf-86f1-41af-91ab-2d7cd011db47'
+              ProductStateField: $(Tag)
+
+          - pwsh: >
```

Comment:
It looks like this step is in the foreach loop which means we will do the entire list for every item in the list. 

Are we going to be able to do this tagging still?

Reply:
it's only looking into each package once